src/averageAges.js

```diff
@@ -56,6 +87,22 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const onlyWomen = people.filter(({ sex }) => sex === 'f');
+  const children = people.filter(({ mother, sex }) => {
+    const checkMother = onlyWomen.some(women => mother === women.name);
+    const son = sex === 'm';
+
+    return onlyWithSon ? checkMother && son : checkMother;
+  }
+  );
+
+  const averageAges = children.reduce((acc, { born, mother }) => {
+    const isMother = onlyWomen.find((women) => mother === women.name);
```

Comment:
Consider renaming `isMother` to `mother` for better naming consistency. 🤔

Reply:
Rename `{ born, mother }` above to `{ born, motherName }` as variable represents name, actually